### PR TITLE
Sample module

### PR DESCRIPTION
**Pull request by YanaBr**
Sample module with the components using all AEM Toolkit annotations, custom annotation, custom dependsOn actions and documentation.

### CODE REVIEW

File: aem-authoring-toolkit-samples/toolkit-samples-package/src/main/content/META-INF/vault/filter.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<workspaceFilter version="1.0">
+    <filter root="/apps/authoring-toolkit/samples"/>
+    <filter root="/apps/acs-list"/>
+    <filter root="/content/aem-authoring-toolkit-samples"/>
+    <filter root="/content/acs-list"/>
+    <filter root="/content/cq:tags"/>
+    <filter root="/etc/designs"/>
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-samples/toolkit-samples-package/src/main/content/META-INF/vault/filter.xml]: these filters are too general and can potentially delete already existing user content

File: aem-authoring-toolkit-samples/toolkit-samples-bundle/src/main/java/com/exadel/aem/toolkit/samples/models/AbilitiesComponent.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            },
+            emptyText = "Select ability")
+    @DialogField(
+            name = "./ability",
+            ranking = 2
+    )
+    @ValueMapValue
+    private String ability;
+
+    @DependsOn(query = "@ability === \"magic\"")
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-samples/toolkit-samples-bundle/src/main/java/com/exadel/aem/toolkit/samples/models/AbilitiesComponent.java]: can it be simplified to `query = "@ability ==='magic'"`?

File: aem-authoring-toolkit-samples/toolkit-samples-package/src/main/content/jcr_root/apps/authoring-toolkit/samples/components/content/abilities-component/content/elements-list.html [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<template data-sly-template.elements="${@ data}">
+    <button id="warrior-abilities-button">ELEMENTS</button>
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-samples/toolkit-samples-package/src/main/content/jcr_root/apps/authoring-toolkit/samples/components/content/abilities-component/content/elements-list.html]: is it safe to use `id`s in the components? what if the component is added to the page multiple times?

File: aem-authoring-toolkit-samples/toolkit-samples-package/src/main/content/jcr_root/apps/authoring-toolkit/samples/components/content/abilities-component/clientlib/authoring/altVisibility.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+/*
```
Comment (ala-n) [inline aem-authoring-toolkit-samples/toolkit-samples-package/src/main/content/jcr_root/apps/authoring-toolkit/samples/components/content/abilities-component/clientlib/authoring/altVisibility.js]: Awesome! 👍 👍 👍

File: aem-authoring-toolkit-samples/toolkit-samples-package/src/main/content/META-INF/vault/filter.xml (hunk 1 above)
Comment (ala-n) [reply aem-authoring-toolkit-samples/toolkit-samples-package/src/main/content/META-INF/vault/filter.xml]: Also it looks like not all of them are needed (like acs-list content, etc.)

File: aem-authoring-toolkit-samples/toolkit-samples-bundle/src/main/java/com/exadel/aem/toolkit/samples/models/WarriorDescriptionComponent.java [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+                    @Characters(name = "Euro sign", entity = "&#x20AC"),
+                    @Characters(name = "Trademark", entity = "&#x2122"),
+                    @Characters(rangeStart = 48, rangeEnd = 70),
+            },
+            formats = {
+                    @ParagraphFormat(tag = "h1", description = "My custom header"),
+                    @ParagraphFormat(tag = "h2", description = "My custom subheader")
+            }
+    )
+    @DialogField(name = "./description")
```
Comment (ala-n) [inline aem-authoring-toolkit-samples/toolkit-samples-bundle/src/main/java/com/exadel/aem/toolkit/samples/models/WarriorDescriptionComponent.java]: We can add InplaceEdditing for that field to show power of @Extend functionality

File: aem-authoring-toolkit-samples/toolkit-samples-bundle/src/main/java/com/exadel/aem/toolkit/samples/models/WarriorDescriptionComponent.java [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+                    Arrays.stream(fruits).
+                            map(fruit -> tagManager.resolve(fruit).getTitle()).
+                            toArray(String[]::new)));
+        }
+        return "He doesn't like fruits";
+    }
+
+    public String getInitDescription() {
+        return
+                String.format("%s was born on the cold but bright day %s. He is %s. %s, and %s",
```
Comment (ala-n) [inline aem-authoring-toolkit-samples/toolkit-samples-bundle/src/main/java/com/exadel/aem/toolkit/samples/models/WarriorDescriptionComponent.java]: Constant for template

File: aem-authoring-toolkit-samples/toolkit-samples-bundle/src/main/java/com/exadel/aem/toolkit/samples/models/WarriorDescriptionComponent.java [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+                            map(film -> film.replaceAll("^.*/(.*)$", "$1")).
+                             toArray(String[]::new)));
+        }
+        return "he doesn't like films";
+    }
+
+    public String getFruits() {
+        if (isLikeFruits && fruits != null) {
+            TagManager tagManager = resourceResolver.adaptTo(TagManager.class);
+            return ("His favorite fruits: " + String.join(", ",
```
Comment (ala-n) [inline aem-authoring-toolkit-samples/toolkit-samples-bundle/src/main/java/com/exadel/aem/toolkit/samples/models/WarriorDescriptionComponent.java]: Split actions please + Collectors.joining

File: aem-authoring-toolkit-samples/toolkit-samples-bundle/src/main/java/com/exadel/aem/toolkit/samples/models/WarriorDescriptionComponent.java [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+
+        if (birthday != null) {
+            return birthday;
+        }
+        return "29.03.2000";
+    }
+
+    public String getFilms() {
+        if (isLikeFilms && films != null) {
+            return ("his favorite film genres: " + String.join(", ",
```
Comment (ala-n) [inline aem-authoring-toolkit-samples/toolkit-samples-bundle/src/main/java/com/exadel/aem/toolkit/samples/models/WarriorDescriptionComponent.java]: Same as bellow in getFruits

File: aem-authoring-toolkit-samples/toolkit-samples-package/src/main/content/jcr_root/apps/authoring-toolkit/samples/components/content/warrior-description-component/clientlib/authoring/namespaceFilter.js [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+Custom dependsOn action for @Autocomplete.
+The action is intended for dynamic changing tags scope depending on the Warrior component
+(that is the main container for other components in the samples module) color theme
+*/
+
+(function (Granite, $, DependsOn) {
+
+    'use strict';
+
+    const PORT = 'http://localhost:4502';
```
[the remark this answers is not in the document]
Comment (YanaBr) [reply aem-authoring-toolkit-samples/toolkit-samples-package/src/main/content/jcr_root/apps/authoring-toolkit/samples/components/content/warrior-description-component/clientlib/authoring/namespaceFilter.js]: Doesn't work without Port

File: aem-authoring-toolkit-samples/toolkit-samples-core/src/main/java/com/exadel/aem/toolkit/samples/models/fieldsets/ProductsFieldSet.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    @Checkbox
+    @DialogField(label = "Cheese")
+    @ValueMapValue
+    private boolean cheese;
+
+    public boolean getMilk() { return milk; }
+    public boolean getCheese() { return cheese; }
+
+    private String getStringMilk() {
+        return BooleanUtils.isNotTrue(milk) ? "" : "milk, ";
```
Comment (smiakchilo) [inline aem-authoring-toolkit-samples/toolkit-samples-core/src/main/java/com/exadel/aem/toolkit/samples/models/fieldsets/ProductsFieldSet.java]: Umm, why not `return milk ? "milk, " : ""`? These are declared as primitive booleans anyway

File: aem-authoring-toolkit-samples/toolkit-samples-ui.apps/index.js [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+// Environment Settings
+//-----------------------------------
+const workingDirs = [];
+
+["./src/main/content/jcr_root"].forEach(function(item) {
+    let dirPath = path.resolve(__dirname, item);
+    workingDirs.push(dirPath);
+});
+
+// console.log(workingDirs);
```
Comment (smiakchilo) [inline aem-authoring-toolkit-samples/toolkit-samples-ui.apps/index.js]: Please clear out commented code lines  before publishing to production branch

File: aem-authoring-toolkit-samples/toolkit-samples-ui.apps/src/main/content/jcr_root/apps/authoring-toolkit/samples/components/content/dungeons-component/dungeons-component.html [hunk 12]
```diff
@@ -0,0 +1,15 @@
+<sly data-sly-use.data="com.exadel.aem.toolkit.samples.models.DungeonsComponent"></sly>
+<sly data-sly-use.clientlib="/libs/granite/sightly/templates/clientlib.html"></sly>
+<sly data-sly-call="${clientlib.all @ categories='authoring-toolkit.samples.components'}"></sly>
```
Comment (smiakchilo) [inline aem-authoring-toolkit-samples/toolkit-samples-ui.apps/src/main/content/jcr_root/apps/authoring-toolkit/samples/components/content/dungeons-component/dungeons-component.html]: [Cosmetic] Please make sure HTL options are introduced same way everywhere (and it is better with a whitespace after `@ `)

File: aem-authoring-toolkit-samples/toolkit-samples-ui.apps/src/main/content/jcr_root/apps/authoring-toolkit/samples/components/content/dungeons-component/dungeons-component.html [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+
+<div class="dungeon-wrap">
+    <span class="dungeon-title">Dungeon</span>
+    <div class="dungeon-text-wrap">
+        <span class="dungeon-text">
+            <span>Once upon a time in the </span><span>${data.dungeonDescription @context='text'}</span>
```
Comment (smiakchilo) [inline aem-authoring-toolkit-samples/toolkit-samples-ui.apps/src/main/content/jcr_root/apps/authoring-toolkit/samples/components/content/dungeons-component/dungeons-component.html]: Do we need several `<span>`s side-by-side here? Will a space be visible between `..his adventure.` and `There are some..`?

File: aem-authoring-toolkit-samples/toolkit-samples-ui.apps/src/main/content/jcr_root/apps/authoring-toolkit/samples/components/content/warrior-component/warrior-component.html [hunk 14]
```diff
@@ -0,0 +1,12 @@
+<sly data-sly-use.data="com.exadel.aem.toolkit.samples.models.WarriorComponent"></sly>
+<sly data-sly-use.clientlib="/libs/granite/sightly/templates/clientlib.html"></sly>
+<sly data-sly-call="${clientlib.css @ categories='authoring-toolkit.samples.components'}"></sly>
+
+<div class="warrior-component-wrap ${data.colorTheme}">
+    <h1 class  ="warrior-component-title">${data.title || "Make your warrior" @context='text'}</h1>
```
Comment (smiakchilo) [inline aem-authoring-toolkit-samples/toolkit-samples-ui.apps/src/main/content/jcr_root/apps/authoring-toolkit/samples/components/content/warrior-component/warrior-component.html]: I'd rather say, "Compose", or "Create"

File: aem-authoring-toolkit-samples/toolkit-samples-core/src/main/java/com/exadel/aem/toolkit/samples/models/WarriorDescriptionComponent.java [hunk 15]
```diff
@@ -0,0 +1,271 @@
...
+@DependsOnTab(tabTitle = WarriorDescriptionComponent.TAB_FILMS, query = "@isLikeFilms")
+@Model(adaptables = Resource.class, defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL)
+public class WarriorDescriptionComponent {
+
+    static final String TAB_MAIN = "Main info";
+    static final String TAB_TASTES = "Tastes";
+    static final String TAB_FRUIT = "Fruit";
+    static final String TAB_FILMS = "Films";
+
+    private static final String DESCRIPTION_TEMPLATE = "%s was born on the cold but bright day %s. He would %s. %s, and %s";
```
[the remark this answers is not in the document]
Comment (YanaBr) [reply aem-authoring-toolkit-samples/toolkit-samples-core/src/main/java/com/exadel/aem/toolkit/samples/models/WarriorDescriptionComponent.java]: @smiakchilo, I was thinking of a distant cold country in which sunny bright days are still possible)

File: aem-authoring-toolkit-samples/toolkit-samples-core/src/main/java/com/exadel/aem/toolkit/samples/models/HomelandComponent.java [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+            return StringUtils.EMPTY;
+        }
+
+        public String getWarriorName() {
+            String warriorName = null;
+
+            Resource resource = resourceResolver.getResource(PathConstants.WARRIOR_COMPONENT_PATH);
+
+            if (resource != null) {
+                ValueMap resourceValueMap = resource.getValueMap();
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-samples/toolkit-samples-core/src/main/java/com/exadel/aem/toolkit/samples/models/HomelandComponent.java]: (minor) `resourceValueMap` is not used

File: aem-authoring-toolkit-samples/toolkit-samples-core/src/main/java/com/exadel/aem/toolkit/samples/models/WarriorDescriptionComponent.java [hunk 17]
```diff
@@ -0,0 +1,268 @@
...
+                .map(Resource::getParent)
+                .map(Resource::getPath)
+                .orElse("");
+        valueMap.put("parentPath", parentPath);
+        resource.getResourceResolver().commit();
+    }
+
+    public String getWarriorName() {
+        String warriorName;
+        Resource parentResource = Optional.ofNullable(resource.getParent()).map(Resource::getParent).get();
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-samples/toolkit-samples-core/src/main/java/com/exadel/aem/toolkit/samples/models/WarriorDescriptionComponent.java]: please add `isPresent()` check or `orElse()`

**Pull request merged**